### DISCUSSION TITLE
Is there any way to expand and collapse the child elements

### DISCUSSION THREAD

**Discussion by gopusrikanth04** [xyflow/xyflow#1265] [Q&A]
Hello Team,
 
   Is there any way to expand and collapse the child elements

![image](https://user-images.githubusercontent.com/47174113/121891836-974a7000-cd39-11eb-97db-41b288614fac.png)

if I click on node-1 . node 2 and node 3 should be collapsed and expand.

How to solve this

Thank you

**Comment by Azim-Ahmed** [comment]
no

**Reply by 21bloc** [reply]
lol, thats dumb

**Comment by datoslabs** [comment]
Depending on how you want to render the nodes in "collapsed" state, you can implement it via custom nodes.  The [Contextual Zoom example](https://reactflow.dev/examples/contextual-zoom-features) shows a custom node that alters the node content based on zoom level; you can implement something similar based on `selected` property that's provided to your custom node component to render the nodes in "collapsed" state.

**Comment by SimonSomlai** [comment]
Not sure if this is very new; https://reactflow.dev/examples/hidden/. But it seems like it would do what you're asking 🤔 

> Nodes and edges can be hidden by using the isHidden attribute. This is useful for implementing expandable/collapsible nodes.

**Reply by moklick** [reply]
It's not built-in to hide all children of a node, but as @SimonSomlai pointed out the `isHidden` option can be used to achieve this effect.

**Comment by frankdgcapital** [comment]
Expand and collapse is a pro feature found at [ReactFlow]( https://reactflow.dev/docs/examples/layout/expand-collapse/)

**Comment by SUMANTH-hyphen** [comment]
**Custom Expand and Collapse Feature for ReactFlow Charts (No Premium Edition)**

I have created a custom expand and collapse feature for ReactFlow charts that functions similarly to the premium expand and collapse feature available in the library. While it may not be identical to the premium version, it serves as a viable alternative that demonstrates the possibility of implementing this functionality without the need for a premium edition.

I have added the snippet to codesandbox you can experience it here [https://codesandbox.io/s/dynamic-flow-chart-4494qt](https://codesandbox.io/s/dynamic-flow-chart-4494qt)

**Reply by mayurjadhav2002** [reply]
Thanks This Solution Works

**Reply by Mdfaique** [reply]
Ahh, Now it's saying Link Not found, can u share again

**Reply by mayurjadhav2002** [reply]
The wrong link is added. This is right link - https://codesandbox.io/s/dynamic-flow-chart-4494qt

**Reply by ww1724** [reply]
hero!

**Reply by terrellbm** [reply]
Has anyone figured out how to make the positioning of the roots/parents next to each other when collapsed/in the initial state, and farther away when the tree is expanded? I noticed that when dealing with extremely large trees, the parents become very spread out even in the collapsed state 

Also trying to get grouping to work with expand/collapse here, and running into some difficulties

**Reply by erny03** [reply]
Did you figure out a way to do this? I'm having the same problem

**Reply by terrellbm** [reply]
Nope :\

**Reply by gdh51** [reply]
There are some part should modify if it will not trigger rerender
```ts 
  const hide = (hidden, childEdgeID, childNodeID) => (nodeOrEdge) => {
    if (
      childEdgeID.includes(nodeOrEdge.id) ||
      childNodeID.includes(nodeOrEdge.id)
    )
      nodeOrEdge.hidden = hidden;
```
```diff
- return nodeOrEdge;  change to
+ return { ...nodeOrEdge };
```
```ts
  };
```
